### ISSUE TITLE
Provide feedback to mentors

### ISSUE THREAD

**Issue by gavt13** [learn-and-code/Manbot#9]
First thing to do is think about what we're doing, so today won't be spent coding but instead planning.
We will be thinking about what each function needs to do and how it should work.
It is important not to just think how the bot will use the function but instead create a function that the bot can use but also works standalone.

So here are the deliverables:
1 - How the function will work
2 - How the bot will use the function
3 - What roles will have acess? What channels will it use
4 - What does the function need to push to and pull from the database.

**Comment by SaranVad-1996** [comment]
So the idea behind function #4 is for a user to send a feedback to a specific mentor?
#"MentorName" "message to mentor"
#"Garrett R" "Great Work!"
Something like that?

**Comment by DylanReedy** [comment]
Tokenize() is purpose-agnostic; it's just there to parse a valid message (messages starting with !) Into constituent commands for the bot to interpret

**Comment by gavt13** [comment]
TBH I dont really know the need for this function if you would prefer you can write up a function you think would be more useful

**Comment by DylanReedy** [comment]
Every module needs to parse messages, no?

**Comment by gavt13** [comment]
This isnt about the parsing of input this is designing a function for the bot to use. Messages will be parsed and then sent to the relevant functions

**Comment by SaranVad-1996** [comment]
I don't have some ideas now so I'll continue with this. I'll try to come up with something until its becomes too late.

**Comment by grzegorzgoldapl** [comment]
I think there was miscommunication here. @SaranVad-1996 meant function 4, from the list of 4 features Gavin provided. Adding hash to number on GitHub references issue with that number, which is assigned to @DylanReedy and isn't directly related.

**Comment by SaranVad-1996** [comment]
[FeedBackToMentor.docx](https://github.com/learn-and-code/Manbot/files/1670240/FeedBackToMentor.docx)

**Comment by gavt13** [comment]
Reviewed:
1 - Nice and simple kept the scope small and should be ok to implement.
2 - In future us a google drive link (just saves me a download)
3 - Need to think where this bot will interact with learners (so that mentors dont see who is giving feedback, maybe a channel solely for learners?)
4 - Might be nice to have feedback for an over all project as well like !feedback budgeting "really enjoyed it"
5 - Think about these suggestions and start planning out your classes and methods in English (dont code)

**Comment by SaranVad-1996** [comment]
For point 4, is that a message that goes to the mentor?

**Comment by gavt13** [comment]
I dunno tbh whatever you think might be nice to go to mentors but if youve another idea go for it

**Comment by SaranVad-1996** [comment]
https://docs.google.com/document/d/1VbLS6gTAV5qQoPMzAGYuyFnOfR_CBwhOrc3-7bTUf_M/edit?usp=sharing

**Comment by gavt13** [comment]
**Reviewed:**
1 - All calls to the bot will start with ! so Eg !feedback
2 - We have someone working on making sure the input is correct from the user. If it says !feedback it will call your class. InputTextTokenizer is not an interface? I dont know were you got that from.
3 - Every time you expect the bot and human to interact there will need to be a command. 
Example:
User in general channel - !feedback (or User Pm's Bot !feedback)
Bot PM's User - Who would you like to leave feed back on? Lists mentors (ie 1 - gavin.tynan). Please structure your command like this "!GiveFeedback 1 very helpful"
User PM's - !GiveFeedback 1 good project
Bot PM's Mentor the feedback
4 - You're not planning out the database. You're just meant to state what information you need from the database. Ie I need a list of all the mentors
5 - It's very vague and alot of things are still unclear . This was meant to be more specific talking more about how your Classes would work and what your methods would do so that when you go to code it you are less likely to run into unforeseen problems
6 - In future take a look at what other people are doing and re read what was assigned. If you looked at the issues on git you would have seen someone is working on tokenising the input and if you re read what was assigned, "What does the function need to push to and pull from the database." You would have see you weren't creating the database
**TLDR:** Making sure you are clear on what's expected will save you time in the long run. Be sure to reread assignments and check up on what is being worked on Git.

**Comment by SaranVad-1996** [comment]
How will the feedback function open up? Will someone else write a function that prompts the user to enter or should it be me? So user calls feedback and bot will say "Enter your feed to the appropriate mentor: ". .

**Comment by gavt13** [comment]
The user will call the function is !rate. That will be parsed earlier in the program and then call your function. Your function will then write back to the user or deal with what has been called

**Comment by gavt13** [comment]
Much much better way clearer
